### PR TITLE
IO Exception handling

### PR DESCRIPTION
**Pull request by VivekVinushanth**

### CODE REVIEW

File: components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterUtil.java [hunk 1]
```diff
@@ -185,6 +186,12 @@ public static void makeAsyncAPICall(SecurityEventTokenPayload securityEventToken
             }
             requestBuilder.setBody(jsonString, ContentType.APPLICATION_JSON);
         } catch (IOException | IdentityEventException | ParseException e) {
+            if(e instanceof IdentityEventException) {
+                if (((IdentityEventException) e).getErrorCode().equals(IO_EXCEPTION_ERROR_CODE)) {
```
Comment (JKAUSHALYA) [inline components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterUtil.java:190]: Don't we need to log the exception here if we are not throwing?

Comment (VivekVinushanth) [reply components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterUtil.java:190]: It is being logged at lower level: https://github.com/wso2-extensions/identity-outbound-adapters/blob/main/components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/EventPayloadCryptographyUtils.java#L203

File: components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterUtil.java [hunk 2]
```diff
@@ -185,6 +172,12 @@ public static void makeAsyncAPICall(SecurityEventTokenPayload securityEventToken
             }
             requestBuilder.setBody(jsonString, ContentType.APPLICATION_JSON);
         } catch (IOException | IdentityEventException | ParseException e) {
+            if(e instanceof IdentityEventException) {
+                if (((IdentityEventException) e).getErrorCode().equals(ERROR_RETRIEVING_ENCRYPTION_PUBLIC_KEY.getCode())) {
```
Comment (ThaminduDilshan) [inline components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterUtil.java:176]: Shall we make this other way around?
`ERROR_RETRIEVING_ENCRYPTION_PUBLIC_KEY.getCode().equals(((IdentityEventException) e).getErrorCode())`

File: components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterConstants.java [hunk 3]
```diff
@@ -107,7 +107,11 @@ public enum ErrorMessages {
         ERROR_CREATING_ASYNC_HTTP_CLIENT("65008", "Error while creating the Async HTTP client.",
                 "Server error encountered while creating the Async HTTP Client of WebSub Hub Adapter."),
         TOPIC_DEREGISTRATION_FAILURE_ACTIVE_SUBS("65009", "Error occurred while de-registering topic", "Backend error" +
-                " received from WebSubHub while attempting to de-register topic: %s. Active subscribers: %s.");
+                " received from WebSubHub while attempting to de-register topic: %s. Active subscribers: %s."),
+
```
Comment (kayathiri4) [inline components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterConstants.java:111]:
```suggestion
```

Comment (kayathiri4) [reply components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterConstants.java:111]: Shall we remove line 114 as well?

**Review by kayathiri4 — APPROVED** [review]

File: components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterConstants.java [hunk 4]
```diff
@@ -107,7 +107,9 @@ public enum ErrorMessages {
         ERROR_CREATING_ASYNC_HTTP_CLIENT("65008", "Error while creating the Async HTTP client.",
                 "Server error encountered while creating the Async HTTP Client of WebSub Hub Adapter."),
         TOPIC_DEREGISTRATION_FAILURE_ACTIVE_SUBS("65009", "Error occurred while de-registering topic", "Backend error" +
-                " received from WebSubHub while attempting to de-register topic: %s. Active subscribers: %s.");
+                " received from WebSubHub while attempting to de-register topic: %s. Active subscribers: %s."),
+        ERROR_RETRIEVING_ENCRYPTION_PUBLIC_KEY("65013", "Unable to retrieve event encryption public key",
+                "Unable to retrieve event encryption public key from %s");
```
Comment (ThaminduDilshan) [inline components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterConstants.java:112]:
```suggestion
                "Unable to retrieve event encryption public key from %s.");
```

File: components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/EventPayloadCryptographyUtils.java [hunk 5]
```diff
@@ -70,6 +71,7 @@ public class EventPayloadCryptographyUtils {
     private static final ConcurrentMap<String, DefaultJWKSetCache> cacheMap = new ConcurrentHashMap<>();
     private static final KeyGenerator keyGenerator;
 
+
```
Comment (ThaminduDilshan) [inline components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/EventPayloadCryptographyUtils.java:74]:
```suggestion
```

**Review by ThaminduDilshan — APPROVED** [review]

**Review by ThaminduDilshan — APPROVED** [review]

**Pull request merged**